app/frontend/src/components/Map.tsx

```diff
@@ -69,8 +67,11 @@ export default function Map({
     return { url };
   };
 
+  const mapRef = useRef<mapboxgl.Map>();
   useEffect(() => {
     if (!containerRef.current) return;
+    //don't create a new map if it exists already
+    if (mapRef.current) return;
```

Comment:
This is an "easy" solution, but I'm not sure it's really the most correct solution.

Reply:
Is this because it gets re-run on re-renders? Does this `useEffect` need to run more than once? I guess most of the dependencies are probably messing this up as they are mostly objects/functions...

Reply:
Honestly, I'm not really sure what was happening. I tried making sure all the dependencies were using `useCallback` but it didn't help. I used a dependency tracking hook and it said that map had a first render every time I typed, but in that case, why were more and more maps being added to the same div? Maybe because they originate from outside react?

Reply:
But this is working so 🤷🏻